source/extensions/all_extensions.bzl

```diff
@@ -1,49 +1,31 @@
-load("@bazel_skylib//lib:dicts.bzl", "dicts")
 load("@envoy_build_config//:extensions_build_config.bzl", "EXTENSIONS")
 
-# These extensions are registered using the extension system but are required for the core Envoy build.
-# The map may be overridden by extensions specified in envoy_build_config.
-_required_extensions = {
-    "envoy.common.crypto.utility_lib": "//source/extensions/common/crypto:utility_lib",
-    "envoy.request_id.uuid": "//source/extensions/request_id/uuid:config",
-    "envoy.transport_sockets.tls": "//source/extensions/transport_sockets/tls:config",
-}
-
 # Return the extension cc_library target after select
 def _selected_extension_target(target):
     return target + "_envoy_extension"
 
+def _get_extensions():
+    return {
+        k: v
+        for k, v in EXTENSIONS.items()
+        if v.get("required") or
+           not v.get("builtin")
```

Comment:
there are 3 tags here for different types of extension

1. **required**:
   these are the additional extensions that were listed in `all_extensions.bzl` - they have been moved to `extensions_build_config.bzl` and marked as required
1. **builtin**:
   these were previously only registered in `generate_extension_db` and so were (mostly) not included here - they are therefore excluded unless they are also marked as `required`
1. **core**:
   these were all already listed in `extensions_build_config.bzl` - but were listed here as being `core` - the `core` tag has simply been moved to the build config file